plutus-tx-plugin/test/AsData/Budget/9.6/recordFields.uplc.golden

```diff
@@ -1,60 +1,45 @@
-                                   (\addInteger ->
-                                      addInteger
-                                        (addInteger
-                                           (addInteger
-                                              (addInteger (addInteger x y) z)
-                                              w)
-                                           (force
-                                              (case
-                                                 (lessThanInteger
-                                                    (addInteger y z)
-                                                    (addInteger x w))
-                                                 [ (delay (addInteger x z))
-                                                 , (delay (addInteger y w)) ])))
-                                        (force
-                                           (case
-                                              (lessThanInteger
-                                                 (addInteger (int d) (int d))
-                                                 (addInteger (int d) (int d)))
-                                              [ (delay
-                                                   (addInteger (int d) (int d)))
-                                              , (delay
-                                                   (addInteger
-                                                      (int d)
-                                                      (int d))) ])))
-                                     (\x y -> addInteger x y))
-                                  (\x y ->
-                                     force ifThenElse
-                                       (lessThanInteger x y)
-                                       (constr 0 [])
-                                       (constr 1 [])))
-                               (int d))
-                            (\ds ->
-                               `$mInts`
-                                 ds
-                                 (\ds ds ds ds -> ds)
-                                 (\void -> error)))
-                         (int d))
-                      (\ds ->
-                         `$mInts` ds (\ds ds ds ds -> ds) (\void -> error)))
-                   (int d))
-                (\ds -> `$mInts` ds (\ds ds ds ds -> ds) (\void -> error)))
-             (int d))
-          (\ds -> `$mInts` ds (\ds ds ds ds -> ds) (\void -> error)))
-       (\scrut cont fail ->
+     (\cse ->
+        (\cse ->
+           (\cse ->
+              (\cse ->
+                 (\cse ->
+                    (\lessThanInteger ->
+                       (\addInteger ->
+                          (\cse ->
+                             (\cse ->
+                                addInteger
+                                  (addInteger
+                                     (addInteger
+                                        (addInteger (addInteger cse cse) cse)
```

Comment:
What is going on here? We're not CSE-ing single variables are we? That would definitely not be a win...

Reply:
> We're not CSE-ing single variables are we? 

I had it written in my notes to check specifically for that and then failed to actually do that, it's been a while since I felt like a complete moron so much...

Reply:
Hmm this seems like a bug, since it applies CSE on partially applied `addInteger`, which it isn't supposed to do.

Reply:
Actually not a bug - the partially applied `addInteger` is in fact not a builtin..